### ISSUE TITLE
Array type object smartmatching against an array instance

### ISSUE THREAD

**Issue by vrurg** [rakudo/rakudo#3917]
## The Problem

An `Array` passes a smartmatch against a concrete array instance:

```raku
say Array ~~ [Real]; # True
```

To my view this smartmatch has to result in `False`.

**Comment by vrurg** [comment]
Discovered as a result of failing `Color` module with #3891.

**Comment by vrurg** [comment]
@jnthn This behavior is not specced. Therefore I don't consider this a _language_-labelled problem-solving issue. But anyway may require your resolution.

**Comment by vrurg** [comment]
The cause of this behavior is this line in `List.pm6`:

```raku
    multi method ACCEPTS(List:D: Iterable:U --> True) { }
```

Replacing `True` to `False` fixes `Color` module broken by #3891. Rakudo tests are passing. Spectesting is currently in progress, will check the results tomorrow.

**Comment by jnthn** [comment]
I agree it should come out as `False`.  I'm curious why this candidate is needed at all.

**Comment by lizmat** [comment]
It was done with 385308d359, which was a fix for https://github.com/rakudo/rakudo/issues/1833

**Comment by vrurg** [comment]
Ok, I've got back to this issue and I'm really confused. @lizmat, are you sure that `List ~~ List.new` must be true? If so, then `Int ~~ 42` must be true too. The same reasoning applies to `Slip ~~ Empty`. 

I've changed the `ACCEPT` method to return `False` and the only failing spectest is `S32-basics/warn.t` where it checks out the outcome of `List ~~ List.new`. So, I'm going to make a PR get this done right.
